expression/builtin_compare_vec.go

```diff
@@ -671,11 +671,72 @@ func (b *builtinNEStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column
 }
 
 func (b *builtinGreatestTimeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinGreatestTimeSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalString(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	n := input.NumRows()
+	buf1, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+
+	buf2, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf2)
+
+	src := result
+	arg := buf1
+	dst := buf2
+	sc := b.ctx.GetSessionVars().StmtCtx
+	for j := 1; j < len(b.args); j++ {
+		if err := b.args[j].VecEvalString(b.ctx, input, arg); err != nil {
+			return err
+		}
+		for i := 0; i < n; i++ {
+			if src.IsNull(i) || arg.IsNull(i) {
+				dst.AppendNull()
+				continue
+			}
+			srcStr := src.GetString(i)
+			argStr := arg.GetString(i)
+			srcTime, err := types.ParseDatetime(sc, srcStr)
```

Comment:
The `srcTime` here is the last loop's `argTime`, so actually we parse each `timeStr` twice, which can be optimized.

Reply:
I will fix it.